rasa/cdu/stack/utils.py

```diff
@@ -42,3 +55,34 @@ def top_user_flow_frame(dialogue_stack: DialogueStack) -> Optional[UserFlowStack
         if isinstance(frame, UserFlowStackFrame):
             return frame
     return None
+
+
+def filled_slots_for_active_flow(
+    dialogue_stack: DialogueStack, all_flows: FlowsList
+) -> Set[str]:
+    """Get all slots that have been filled for the current flow.
+
+    Args:
+        tracker: The tracker to get the filled slots from.
+        all_flows: All flows.
+
+    Returns:
+    All slots that have been filled for the current flow.
+    """
+    asked_collect_information = set()
+
+    for frame in reversed(dialogue_stack.frames):
+        if not isinstance(frame, BaseFlowStackFrame):
+            break
```

Comment:
shouldn't this be continue? I assume you wanna skip over all the non base flow stack frames that might be there? (although not sure there would be any?)

Reply:
phew I think it can't: I am not sure e.g. doing a correction in a user flow that isn't at the top works

Reply:
So this is more of a case of an erroneous state? Maybe then let's a comment

Reply:
I thought that is what the second break statement is for. And that also has a comment

There's two in this loop. I am talking here about the upper one, which seems more at a place where I would expect a continue. And it doesn't have a comment, and I struggle a bit with interpreting it.

Reply:
mhm I mean this basically collects all questions from all patterns and the latest user flow. 

if there is something else in the middle it breaks and it breaks if it reches the latest user flow.

the first break is to abort when we hit a `chitchat` frame or something in between. not sure if we should continue over that 🤔

Reply:
I mean it must include patterns, because if you have a correction frame in there that poses a collect_information (e.g. to confirm the correction), you need to be able to correct that as well (we have an e2e test for that).

I don't have a strong opinion on flows other than `BaseStackFlowFrame`s happy to change it to continue 🤷

Reply:
I do find it a bit surprising that we are lumping together patterns and user flows. Imagine you have a correction flow and a user flow on the stack. When you call it, you might get a mix of the patterns slots and the user flows slots. I think that's a bit odd. Especially since a chitchat frame by itself constitutes an "active flow" (from the functions name), but a pattern does not.

Reply:
don't think we have a relevant case here at the moment, because all non `BaseStackFlowFrame`s are ended before the next user input.